src/self_driving/pilot_util.cpp

```diff
@@ -12,19 +12,148 @@
 #include "loggers/selfdriving_logger.h"
 #include "messenger/messenger.h"
 #include "metrics/metrics_thread.h"
+#include "metrics/metrics_util.h"
+#include "network/postgres/statement.h"
 #include "optimizer/cost_model/trivial_cost_model.h"
 #include "parser/expression/constant_value_expression.h"
 #include "parser/postgresparser.h"
+#include "parser/variable_set_statement.h"
+#include "planner/plannodes/abstract_plan_node.h"
 #include "self_driving/forecast/workload_forecast.h"
 #include "self_driving/model_server/model_server_manager.h"
+#include "self_driving/modeling/operating_unit.h"
 #include "self_driving/pilot/pilot.h"
+#include "settings/settings_manager.h"
 #include "traffic_cop/traffic_cop_util.h"
 #include "transaction/transaction_manager.h"
 
 namespace noisepage::selfdriving {
 
+void PilotUtil::ApplyAction(common::ManagedPointer<Pilot> pilot, const std::string &sql_query,
+                            catalog::db_oid_t db_oid) {
```

Comment:
Add some documentation to this function as well. In general, if the function is not super simple, we should always document the code logic along the way. So that other people (other than you and me) who happen to need to touch this function can know what's going on.

Reply:
Sure, I put it in the header file.

Reply:
Eh, I was actually referring to the code logic in the cpp file. It's good to add a little bit of explanation along with the code so it's easier for others to understand if they ever need to refactor the implementation.